pkg/controller/clusterprovision/metrics.go

```diff
@@ -2,29 +2,51 @@ package clusterprovision
 
 import (
 	"github.com/prometheus/client_golang/prometheus"
-
 	"sigs.k8s.io/controller-runtime/pkg/metrics"
 )
 
 var (
+	// Declare the metrics which allow optional labels to be added.
+	// They are defined later once the hive config has been read.
+	metricClusterProvisionsTotal *prometheus.CounterVec
+	metricInstallErrors          *prometheus.CounterVec
+
+	metricInstallFailureSeconds *prometheus.HistogramVec
+	metricInstallSuccessSeconds *prometheus.HistogramVec
+)
+
+func registerMetrics() {
```

Comment:
Unless there's a really good reason to do otherwise, we should make `optionalClusterTypeLabels` a param to this func rather than a global.